include/zephyr/bluetooth/mesh/access.h

```diff
@@ -673,6 +672,23 @@ struct bt_mesh_mod_id_vnd {
 };
 
 /** Abstraction that describes a Mesh Model instance */
+struct bt_mesh_model_data {
+	/* Internal information, mainly for persistent storage */
+	uint8_t  elem_idx;   /* Belongs to Nth element */
+	uint8_t  mod_idx;    /* Is the Nth model in the element */
+	uint16_t flags;      /* Model flags for internal bookkeeping */
+
+#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
+	/* Pointer to the next model in a model extension list. */
+	const struct bt_mesh_model *next;
```

Comment:
I wish we could also move the model extension list to the flash since it also doesn't change at runtime. It could also help to get rid of the static array that we currently need to build composition data page 1: https://github.com/zephyrproject-rtos/zephyr/blob/6c93cbf7b496229172fa9508b561417bf91df0ae/subsys/bluetooth/mesh/access.c#L78

Reply:
year, no plan for this pr.

Reply:
btw, elem_idx and mod_idx also can const.

Reply:
the catually only one variable just left `flags`.

Reply:
I'm thinking about how to initialize mod when defining `mod_idx` and `elem_idx`, but still no ideas and I can't come up with a compatible method.

Reply:
> I'm thinking about how to initialize mod when defining `mod_idx` and `elem_idx`, but still no ideas and I can't come up with a compatible method.

I don't see a good solution either right now. They could be though calculated at runtime whenever they are needed.